django/db/models/aggregates.py

```diff
@@ -78,6 +78,14 @@ def __init__(self, expression, **extra):
         output_field = extra.pop('output_field', FloatField())
         super(Avg, self).__init__(expression, output_field=output_field, **extra)
 
+    def as_oracle(self, compiler, connection):
+        if self.output_field.get_internal_type() == 'DurationField':
+            expression = self.get_source_expressions()[0]
+            from django.db.backends.oracle.functions import IntervalToSeconds, SecondsToInterval
+            return compiler.compile(
```

Comment:
one line would be fine here I think, or I prefer:
```
return compiler.compile(
    SecondsToInterval(Avg(IntervalToSeconds(expression)))
)
```
so it's easier to balance the parenthesis